openesdh-repo/src/main/java/dk/openesdh/repo/services/activities/ActivitiesFeedMailMessageResolver.java

```diff
@@ -0,0 +1,135 @@
+            }
+            Map<String, Object> activity = (Map<String, Object>) ((SimpleHash) arguments.get(0)).toMap();
+            return getResolver(activity)
+                    .map(resolver -> resolver.resolve(activity))
+                    .orElse("No resolver for activity type " + activityType(activity));
+        } catch (Exception ex) {
+            logger.error("Error resolving activity message: ", ex);
+            throw ex;
+        }
+    }
+
+    public void registerMesageResolver(String activityType, MessageResolver resolver) {
+        resolvers.put(activityType, resolver);
+    }
+
+    private Optional<MessageResolver> getResolver(Map<String, Object> activity) {
+        return Optional.ofNullable(resolvers.get(activityType(activity)));
+    }
+
+    private void initResolvers() {
+        resolvers.put(ACTIVITY_TYPE_CASE_UPDATE, new MessageResolver(MODIFIER_DISPLAY_NAME, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_MEMBER_ADD, new MessageResolver(MODIFIER_DISPLAY_NAME, MEMBER, ROLE, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_MEMBER_REMOVE, new MessageResolver(MODIFIER_DISPLAY_NAME, MEMBER, ROLE, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_DOCUMENT_UPLOAD, new MessageResolver(MODIFIER_DISPLAY_NAME, DOC_TITLE, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_DOCUMENT_NEW_VERSION_UPLOAD, new MessageResolver(MODIFIER_DISPLAY_NAME, DOC_TITLE, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_DOCUMENT_ATTACHMENT_UPLOAD, new MessageResolver(MODIFIER_DISPLAY_NAME, ATTACHMENT_TITLE, DOC_TITLE, CASE_TITLE, CASE_ID));
+        resolvers.put(ACTIVITY_TYPE_CASE_DOCUMENT_ATTACHMENT_NEW_VERSION_UPLOAD, new MessageResolver(MODIFIER_DISPLAY_NAME, ATTACHMENT_TITLE, DOC_TITLE, CASE_TITLE, CASE_ID));
+        
+        MessageResolver workflowMessageResolver = new MessageResolver(MODIFIER_DISPLAY_NAME, WORKFLOW_DESCRIPTION, CASE_TITLE, CASE_ID);
+        resolvers.put(ACTIVITY_TYPE_CASE_WORKFLOW_START, workflowMessageResolver);
+        resolvers.put(ACTIVITY_TYPE_CASE_WORKFLOW_CANCEL, workflowMessageResolver);
+        resolvers.put(ACTIVITY_TYPE_CASE_WORKFLOW_TASK_END, workflowMessageResolver);
+        resolvers.put(ACTIVITY_TYPE_CASE_WORKFLOW_TASK_APPROVE, workflowMessageResolver);
+        resolvers.put(ACTIVITY_TYPE_CASE_WORKFLOW_TASK_REJECT, workflowMessageResolver);
+    }
+
+    private static String activityType(Map<String, Object> activity) {
+        return (String) activity.get(ACTIVITY_TYPE);
+    }
+
+    public static class MessageResolver {
+
+        private String[] paramNames;
+
+        public MessageResolver(String... paramNames) {
+            this.paramNames = paramNames;
+        }
+
+        public String resolve(Map<String, Object> activity) {
+            Map<String, String> summary = (Map<String, String>) activity.get("activitySummary");
+            List<String> params = Arrays.asList(this.paramNames)
```

Comment:
you can skip collecting to list and later converting to array
stream().map().toArray()